pkg/minio/minio.go

```diff
@@ -0,0 +1,139 @@
+	minioConfig, err := cli.CoreV1().ConfigMaps("kubeless").Get("minio-config", metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("Unable to find Minio configuration. Received: %s", err)
+	}
+	maxFileSize := minioConfig.Data["maxFileSize"]
+	maxSizeQuantity, err := resource.ParseQuantity(maxFileSize)
+	if err != nil {
+		return err
+	}
+	maxSize, _ := maxSizeQuantity.AsInt64()
+	if fileSize > maxSize {
+		return fmt.Errorf("The maximum size of a function is %s", maxFileSize)
+	}
+	return nil
+}
+
+// UploadFunction uploads a file to Minio using as object name file.extension
+// It uses a Kubernetes job to access Minio since we need to use an URL <domain>[:port] (a URL with
+// proxy is not valid)
+func UploadFunction(file, checksum string, fileStats os.FileInfo, cli kubernetes.Interface) (string, error) {
+	err := validateMaxSize(fileStats.Size(), cli)
+	if err != nil {
+		return "", err
+	}
+	minioCredentials := "minio-key"
+	jobName := "upload-file-" + checksum[0:10]
+	fileName := path.Base(file) + "." + checksum
+	var absPath string
+	if !path.IsAbs(file) {
+		cwd, err := os.Getwd()
+		if err != nil {
+			return "", err
+		}
+		absPath = path.Join(cwd, file)
+	} else {
+		absPath = file
+	}
+	job := batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      jobName,
+			Namespace: "kubeless",
+		},
+		Spec: batchv1.JobSpec{
+			Template: v1.PodTemplateSpec{
+				Spec: v1.PodSpec{
+					Volumes: []v1.Volume{
+						{
+							Name: minioCredentials,
+							VolumeSource: v1.VolumeSource{
+								Secret: &v1.SecretVolumeSource{
+									SecretName: minioCredentials,
+								},
+							},
+						},
+						{
+							Name: "func",
+							VolumeSource: v1.VolumeSource{
+								HostPath: &v1.HostPathVolumeSource{
+									Path: absPath,
+								},
+							},
+						},
+					},
+					RestartPolicy: v1.RestartPolicyNever,
+					Containers: []v1.Container{
+						{
+							Name:  "uploader",
+							Image: "minio/mc:RELEASE.2017-10-14T00-51-16Z",
+							VolumeMounts: []v1.VolumeMount{
+								{
+									Name:      minioCredentials,
+									MountPath: "/minio-cred",
+								},
+								{
+									Name:      "func",
+									MountPath: "/" + path.Base(file),
+								},
+							},
+							Command: []string{"sh", "-c"},
+							Args: []string{
+								"mc config host add minioserver http://minio.kubeless:9000 $(cat /minio-cred/accesskey) $(cat /minio-cred/secretkey); " +
+									"mc cp /" + path.Base(file) + " minioserver/functions/" + fileName,
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	_, err = cli.BatchV1().Jobs("kubeless").Create(&job)
+	if err != nil {
+		return "", err
+	}
+	waitErr := waitForCompletedJob(jobName, "kubeless", 120, cli)
+	// Clean up (delete job)
+	err = cli.BatchV1().Jobs("kubeless").Delete(jobName, &metav1.DeleteOptions{})
+	if waitErr != nil {
+		return "", waitErr
+	}
+	if err != nil {
+		return "", err
+	}
+	return "http://minio.kubeless:9000/functions/" + fileName, nil
```

Comment:
I note you have several hardcoded references to "kubeless" namespace in this function. Are we still able to install kubeless into a different namespace?

Reply:
(removed references)